From 85bdc1ee61c08a5cdc99fe28a1c41eebd90fd6fd Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Wed, 26 Nov 2025 17:03:44 +0800
Subject: [PATCH 4/7] app.py: avoid out-of-bounds access

Previously t_idx might be L which will cause out-of-bounds access.
---
 app.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/app.py b/app.py
index a826dbc..d1639ea 100644
--- a/app.py
+++ b/app.py
@@ -474,10 +474,10 @@ def check_video(video):
     t_idx = 0
     target_frames = []
     for target_t in range(1, L_target+1):
-        while target_t / target_fps > original_t[t_idx]:
-            t_idx += 1      # find the first t_idx so that target_t / target_fps <= original_t[t_idx]
-            if t_idx >= L:
-                break
+        while t_idx < L and target_t / target_fps > original_t[t_idx]:
+            t_idx += 1
+        if t_idx >= L:
+            t_idx = L - 1
         target_frames.append(frames[t_idx])
 
     # save video
-- 
2.43.0

